src/command_modules/azure-cli-role/azure/cli/command_modules/role/tests/role_scenario_test.py

```diff
@@ -0,0 +1,26 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+import uuid
+import mock
+from azure.cli.testsdk import ScenarioTest
+
+
+class RoleScenarioTest(ScenarioTest):
+
+    def enable_large_payload(self, size=8192):
+        from azure_devtools.scenario_tests import LargeResponseBodyProcessor
+        large_resp_body = next((r for r in self.recording_processors if isinstance(
+            r, LargeResponseBodyProcessor)), None)
+        if large_resp_body:
+            large_resp_body._max_response_body = size   # pylint: disable=protected-access
+
+    def get_guid_gen_patch(self, guids, test_seam='azure.cli.command_modules.role.custom._gen_guid'):
```

Comment:
This is to ensure we will have fixed guid (used in the url) recorded into the .yaml files for playback later